ci/build_only/build.bash

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
```

Comment:
This looks great! My only comment is that by gpuCI convention we name this file `ci/cpu/build.sh` if that's okay with you.

Reply:
I wasn't quite sure what to do here; we do have some non-CUDA backends that do need to run tests on CPUs. I guess I was imaging we would have:

- `ci/build_only/build.bash`, which would build everything, and down the road would export the test binaries to run on GPU nodes.
- `ci/cpu/build.bash`, which would just build CPU backends and would run their tests.
- `ci/gpu/build.bash`, which would just build GPU backends and run their tests.

But maybe I'm going about this wrong? What do you think?

Reply:
Perhaps we should have just the two configurations, and have them be fully parameterized? @allisonvacanti, could we perhaps set up our CMake configuration so that CTest only runs GPU tests if the node has GPUs supported? We wouldn't need to detect this - we could have a CMake variable 'THRUST_RUN_CUDA_TESTS', or something like that.

Then we would have:

- `ci/cpu/build.bash`: Builds a parameterized set of configurations, then runs all tests (expect CUDA tests), and packages up the tests for redistribution.
- `ci/gpu/build.bash`: Builds a parameterized set of configurations, then runs all tests.

Reply:
> could we perhaps set up our CMake configuration so that CTest only runs GPU tests if the node has GPUs supported

No need to modify the CMake config for this -- we can do that today with a -R [regex] option to ctest. All GPU tests can be run with `ctest -R "^cub|^thrust.*cuda"`.

Similarly, all CPU tests can be run with the -E [exclude regex] option `ctest -E "^cub|^thrust.*cuda"`.

If we wanted to simplify this we could explicitly tag the tests with `GPU` and `CPU` labels and run them with `ctest -L GPU`, etc, too.

Building today can also be targeted by listing out some per-`host.device.dialect` meta targets in the scripts:

```
# Build all cuda targets:
ninja thrust.cpp.cuda.cpp11.all \
      thrust.cpp.cuda.cpp14.all \
      thrust.cpp.cuda.cpp17.all \
      thrust.omp.cuda.cpp11.all \
      [ ... ] \
      thrust.tbb.cuda.cpp11.all \
      [ ... ] \
      cub.all
```

Similar to the ctest labels, it'd be trivial to add `thrust.gpu.all`, `thrust.cpu.all` metatargets that would simplify this if needed.